WalletWasabi/Crypto/Strobe.cs

```diff
@@ -0,0 +1,318 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace WalletWasabi.Crypto
+{
+	// https://strobe.sourceforge.io/papers/strobe-20170130.pdf
+	// Based on Merlin framework small implementation: https://doc-internal.dalek.rs/src/merlin/strobe.rs.html
+	public class Strobe128
```

Comment:
also implement `IClonable`?

We will need some way of copying strobe states in order to do synthetic nonces, and the rust version also provides a clone method (derived automatically by the rust compiler):

https://github.com/dalek-cryptography/merlin/blob/1ed350bbc1d65f0a0697e0c20c48e11ec172c6ff/src/strobe.rs#L33-L39

Reply:
We need a `Clone` method but [Microsoft recommends not implement `ICloneable`](https://docs.microsoft.com/en-us/dotnet/api/system.icloneable.clone?view=netcore-3.1) basically because it is a very bad interface.

Reply:
I've created the `MakeCopy()` method because `Clone()` is known to return `object` and suggests we should implement `ICloneable` when we don't want to do it.